device/core/test/_client_test.js

```diff
@@ -110,32 +99,50 @@ describe('Client', function () {
 
     /*Tests_SRS_NODE_DEVICE_CLIENT_16_030: [The fromSharedAccessSignature method shall return a new instance of the Client object] */
     it('returns an instance of Client', function () {
-      var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net/devices/deviceId&sig=s1gn4tur3&se=1454204843"';
       var client = Client.fromSharedAccessSignature(sharedAccessSignature, FakeTransport);
       assert.instanceOf(client, Client);
     });
 
-    it('create a correct config when sr is not URI-encoded', function () {
-      var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net/devices/deviceId&sig=s1gn4tur3&se=1454204843"';
-      var DummyTransport = function (config) {
-        this.on = function() {};
-        assert.strictEqual(config.host, 'hubName.azure-devices.net');
-        assert.strictEqual(config.deviceId, 'deviceId');
-        assert.strictEqual(config.sharedAccessSignature, sharedAccessSignature);
-      };
-      Client.fromSharedAccessSignature(sharedAccessSignature, DummyTransport);
+    it('creates a SharedAccessSignatureAuthorizationProvider and passes it to the transport', function (testCallback) {
+      var client = Client.fromSharedAccessSignature(sharedAccessSignature, function (authProvider) {
+        assert.instanceOf(authProvider, SharedAccessSignatureAuthenticationProvider);
+        testCallback();
+      });
     });
+  });
 
-    it('create a correct config when sr is URI-encoded', function () {
-      var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net%2Fdevices%2FdeviceId&sig=s1gn4tur3&se=1454204843"';
-      var DummyTransport = function (config) {
-        this.on = function() {};
-        assert.strictEqual(config.host, 'hubName.azure-devices.net');
-        assert.strictEqual(config.deviceId, 'deviceId');
-        assert.strictEqual(config.sharedAccessSignature, sharedAccessSignature);
-      };
-      Client.fromSharedAccessSignature(sharedAccessSignature, DummyTransport);
+  describe('#fromAuthenticationProvider', function () {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_089: [The `fromAuthenticationProvider` method shall throw a `ReferenceError` if the `authenticationProvider` argument is falsy.]*/
```

Comment:
>falsy [](start = 159, length = 5)

nit: should the requirement be !falsy, or should the requirement be that it must be a function?

Reply:
not sure what you mean. we usually enforce arguments like that but not type of arguments.

---
In reply to: [157076086](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157076086) [](ancestors = 157076086)